best-practices/largescale-deep-learning/Training/ViT-Pretrain/README.md

```diff
@@ -0,0 +1,106 @@
+[![smoke](https://github.com/Azure/azureml-examples/workflows/smoke/badge.svg)](https://github.com/Azure/azureml-examples/actions/workflows/smoke.yml)
+[![Python code style: black](https://img.shields.io/badge/code%20style-black-000000.svg)](https://github.com/psf/black)
+[![license: MIT](https://img.shields.io/badge/License-MIT-purple.svg)](LICENSE)
+# **ViT Pretraining**
+
+The main task we consider in this example is pretraining the vision transformer base (ViTB) model on [ImageNet-1k](https://www.image-net.org/). Larger datasets like ImageNet-22k will likely lead to better model quality, but here we focus on making the training process as efficient as possible and we identify the techniques to achieve this goal. We also trained ViTB on [PANDA](https://www.kaggle.com/c/prostate-cancer-grade-assessment/data), a medical dataset with high resolution images, and synthetic datasets similar to it.
+
+## **General Tips and Tricks**
+
+Based on general best practices for Computer Vision model training and our experimentation on Azure, we synthesized the following guidelines:
+
+1. Set the batch size so that GPU memory consumption is maximized.
+2. Set the shared memory size to a large value, e.g. 50% of the total CPU memory available, to make sure dataloaders have enough memory. Please see the [launcher.py](launcher.py) file in this folder for a code snippet (look for "shm_size").
+3. If network traffic does not exceed the network bandwidth available to the machine, mount the dataset's datastore path and disable caching. This has virtually the same performance as downloading the dataset on the machine when this option is available (disk large enough).
+4. In general, the optimal number of workers for data loading can be found by starting with the number of CPU cores and increasing until there is no noticeable improvement. We have seen that for [NCv3 machines](https://learn.microsoft.com/en-us/azure/virtual-machines/ncv3-series) a good starting point is 6, but for [NCA100v4 machines](https://learn.microsoft.com/en-us/azure/virtual-machines/nc-a100-v4-series) a good starting point is 12.
+5. Scale up before scaling out, e.g. use a Standard_NC24s_v3 with 4 GPUs instead of 4 Standard_NC6s_v3's with 1 GPU each [NCv3 machines](https://learn.microsoft.com/en-us/azure/virtual-machines/ncv3-series). Synchronizing weights on one machine with 4 GPUs (Standard_NC24s_v3) is faster than synchronizing weights across the network between 4 machines with one GPU each (Standard_NC6s_v3).
+6. Use [InfiniBand]#(hardware) whenever possible, as it enables near linear scaling.
```

Comment:
Can we provide a link to Azure infiniband VMs available?

Reply:
We have the link in the Hardware section, which point 6 links to. That section also includes details about Infiniband, so keeping things as they are here.

Reply:
got it. If it provides a link to hardware section, thats good enough.